### ISSUE TITLE
Problem with the genome-nexus service.

### ISSUE THREAD

**Issue by wojgra** [codingsoo/ARAT-RL#4]
Dear Myeongsoo Kim,

I am attempting to reproduce the ARAT-RL research (python run.py arat-rl), but I am encountering an issue with the genome-nexus service crashing. I am using a Google Cloud E2 machine with a 24-core CPU and 96 GB of RAM, running Ubuntu 20.04 LTS. Could I please request assistance in addressing this matter?

I've saved the logs to a file, where it says `Port 50110 was already in use.` After running the `netstat -tuln `command, this port is not in use. Even after changing the port in the `application.properties` file, the same error persists. If I start the genome-nexus service individually, it works fine, but there's an issue when running it with the `python run.py arat-rl` command.

Before using the `python run.py arat-rl` command:
~/ARAT-RL$ netstat -tuln
```
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State      
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:33817         0.0.0.0:*               LISTEN     
tcp6       0      0 :::22                   :::*                    LISTEN     
udp        0      0 127.0.0.53:53           0.0.0.0:*                          
udp        0      0 10.186.0.5:68           0.0.0.0:*                          
udp        0      0 127.0.0.1:323           0.0.0.0:*                          
udp6       0      0 ::1:323                 :::*         
```
After using the `python run.py arat-rl` command:
~/ARAT-RL$ netstat -tuln
```
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State      
tcp        0      0 0.0.0.0:3306            0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:27018           0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:27019           0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:33817         0.0.0.0:*               LISTEN     
tcp6       0      0 :::50115                :::*                    LISTEN     
tcp6       0      0 :::50117                :::*                    LISTEN     
tcp6       0      0 :::50118                :::*                    LISTEN     
tcp6       0      0 :::50100                :::*                    LISTEN     
tcp6       0      0 :::50102                :::*                    LISTEN     
tcp6       0      0 :::50106                :::*                    LISTEN     
tcp6       0      0 :::50108                :::*                    LISTEN     
tcp6       0      0 :::50111                :::*                    LISTEN     
tcp6       0      0 :::3306                 :::*                    LISTEN     
tcp6       0      0 :::22                   :::*                    LISTEN     
tcp6       0      0 :::11090                :::*                    LISTEN     
tcp6       0      0 :::11080                :::*                    LISTEN     
tcp6       0      0 :::11030                :::*                    LISTEN     
tcp6       0      0 :::11010                :::*                    LISTEN     
tcp6       0      0 :::11020                :::*                    LISTEN     
tcp6       0      0 :::11060                :::*                    LISTEN     
tcp6       0      0 :::11070                :::*                    LISTEN     
tcp6       0      0 :::11040                :::*                    LISTEN     
tcp6       0      0 :::11000                :::*                    LISTEN     
tcp6       0      0 127.0.0.1:50101         :::*                    LISTEN     
tcp6       0      0 :::30100                :::*                    LISTEN     
tcp6       1      0 :::30101                :::*                    LISTEN     
tcp6       0      0 :::30102                :::*                    LISTEN     
tcp6       0      0 :::30106                :::*                    LISTEN     
tcp6       0      0 :::30108                :::*                    LISTEN     
tcp6       0      0 :::30110                :::*                    LISTEN     
tcp6       0      0 :::30111                :::*                    LISTEN     
tcp6       0      0 :::30115                :::*                    LISTEN     
tcp6       0      0 :::30117                :::*                    LISTEN     
tcp6       0      0 :::30118                :::*                    LISTEN     
tcp6       0      0 :::27018                :::*                    LISTEN     
tcp6       0      0 :::27019                :::*                    LISTEN     
udp        0      0 127.0.0.53:53           0.0.0.0:*                          
udp        0      0 10.186.0.5:68           0.0.0.0:*                          
udp        0      0 127.0.0.1:323           0.0.0.0:*                          
udp6       0      0 ::1:323                 :::*   
```

Saved logs of the genome-service:

```
2024-05-12 19:24:56.070  INFO 3909 --- [           main] org.mongodb.driver.connection            : Opened connection [connectionId{localValue:2, serverValue:2}] to 127.0.0.1:27018
2024-05-12 19:24:57.147  INFO 3909 --- [           main] m.c.i.MongoPersistentEntityIndexResolver : Found cycle for field 'children' in type 'TumorType' for path 'children -> children'
2024-05-12 19:24:57.149  INFO 3909 --- [           main] m.c.i.MongoPersistentEntityIndexResolver : Found cycle for field 'children' in type 'TumorType' for path 'children -> children'
2024-05-12 19:24:57.150  INFO 3909 --- [           main] m.c.i.MongoPersistentEntityIndexResolver : Found cycle for field 'children' in type 'TumorType' for path 'children -> children'
2024-05-12 19:24:57.165  INFO 3909 --- [           main] m.c.i.MongoPersistentEntityIndexResolver : Found cycle for field 'children' in type 'TumorType' for path 'children -> children'
2024-05-12 19:24:57.166  INFO 3909 --- [           main] m.c.i.MongoPersistentEntityIndexResolver : Found cycle for field 'children' in type 'TumorType' for path 'children -> children'
2024-05-12 19:24:57.168  INFO 3909 --- [           main] m.c.i.MongoPersistentEntityIndexResolver : Found cycle for field 'children' in type 'TumorType' for path 'children -> children'
2024-05-12 19:25:11.240  INFO 3909 --- [           main] o.c.g.s.internal.SignalQueryServiceImpl  : Building transcript to Uniprot id map
2024-05-12 19:25:23.012  INFO 3909 --- [           main] o.c.g.s.internal.SignalQueryServiceImpl  : Finished building transcript to Uniprot id map
2024-05-12 19:25:23.420  INFO 3909 --- [           main] o.c.g.s.internal.SignalQueryServiceImpl  : Building Signal index
2024-05-12 19:26:17.403  INFO 3909 --- [           main] o.c.g.s.internal.SignalQueryServiceImpl  : Finished building Signal index
2024-05-12 19:26:18.382  INFO 3909 --- [           main] o.s.b.a.e.web.EndpointLinksResolver      : Exposing 2 endpoint(s) beneath base path '/actuator'
2024-05-12 19:26:18.537  INFO 3909 --- [           main] pertySourcedRequestMappingHandlerMapping : Mapped URL path [/v2/api-docs] onto method [springfox.documentation.swagger2.web.Swagger2Controller#getDocumentation(String, HttpServletRequest)]
2024-05-12 19:26:18.712  INFO 3909 --- [           main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
2024-05-12 19:26:18.838  INFO 3909 --- [           main] o.s.b.a.w.s.WelcomePageHandlerMapping    : Adding welcome page: class path resource [static/index.html]
2024-05-12 19:26:19.153  INFO 3909 --- [           main] d.s.w.p.DocumentationPluginsBootstrapper : Context refreshed
2024-05-12 19:26:19.199  INFO 3909 --- [           main] d.s.w.p.DocumentationPluginsBootstrapper : Found 2 custom documentation plugin(s)
2024-05-12 19:26:19.376  INFO 3909 --- [           main] s.d.s.w.s.ApiListingReferenceScanner     : Scanning for api listing references
2024-05-12 19:26:20.520  INFO 3909 --- [           main] s.d.s.w.s.ApiListingReferenceScanner     : Scanning for api listing references
2024-05-12 19:26:20.668  INFO 3909 --- [           main] .d.s.w.r.o.CachingOperationNameGenerator : Generating unique operation named: fetchGeneXrefsGET_1
2024-05-12 19:26:20.865  INFO 3909 --- [           main] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]
2024-05-12 19:26:20.873  WARN 3909 --- [           main] o.a.c.loader.WebappClassLoaderBase       : The web application [ROOT] appears to have started a thread named [cluster-ClusterId{value='664117865e268f5698c6b5f2', description='null'}-127.0.0.1:27018] but has failed to stop it. This is very likely to create a memory >
 sun.misc.Unsafe.park(Native Method)
 java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
 java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
 com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.waitForSignalOrTimeout(DefaultServerMonitor.java:229)
 com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.waitForNext(DefaultServerMonitor.java:210)
 com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.run(DefaultServerMonitor.java:157)
 java.lang.Thread.run(Thread.java:750)
2024-05-12 19:26:20.874  WARN 3909 --- [           main] o.a.c.loader.WebappClassLoaderBase       : The web application [ROOT] appears to have started a thread named [CleanCursors-1-thread-1] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 sun.misc.Unsafe.park(Native Method)
 java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
 java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
 java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1093)
 java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:809)
 java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
 java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
 java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
 java.lang.Thread.run(Thread.java:750)
2024-05-12 19:26:20.884  INFO 3909 --- [           main] ConditionEvaluationReportLoggingListener :

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2024-05-12 19:26:20.888 ERROR 3909 --- [           main] o.s.b.d.LoggingFailureAnalysisReporter   :

***************************
APPLICATION FAILED TO START
***************************

Description:

Web server failed to start. Port 50110 was already in use.

Action:

Identify and stop the process that's listening on port 50110 or configure this application to listen on another port.
```

**Comment by codingsoo** [comment]
Hi @wojgra 

I just ran the setup script and executed python run.py arat-rl in Google EC2 machine, and it worked well without a port crash. Could you check if your old sessions are running with tmux ls? If there are old sessions, you can remove them all using the tmux kill-server command. Let me know if this does not work!

**Comment by wojgra** [comment]
Hi  @codingsoo, 
Thank you for your response. Yes, I make sure that there is no tmux session. Unfortunately, I'm encountering this issue. However, I worked around it by starting the service manually (commenting out the genome-nexus part in run_service.py) and then running the command python run.py arat-rl, and this way it works. I also noticed that using the command python run.py restler does not cause this issue.
